docs/docs/lifting-state-up.md

```diff
@@ -0,0 +1,223 @@
+---
+id: lifting-state-up
+title: Lifting State Up
+permalink: docs/lifting-state-up.html
+prev: state-and-lifecycle.html
+next: composition-vs-inheritance.html
+---
+
+Often, several components need to reflect the same changing data. We recommend to lift the shared state up to their closest common ancestor. Let's see how this works in action.
+
+In this section, we will create a temperature calculator that calculates whether the water would boil at a given temperature.
+
+We will start with a component called `BoilingVerdict`. It accepts the `celsius` temperature as a prop, and prints whether it is enough to boil the water:
+
+```js{3,5}
+function BoilingVerdict(props) {
+  if (props.celsius >= 100) {
+    return <p>The water would boil.</p>;
+  } else {
+    return <p>The water would not boil.</p>;
+  }
+}
+```
+
+Next, we will create a component called `Calculator`. It renders an `<input>` that lets you enter the temperature, and keeps its value in `this.state.celsius`.
+
+Additionally it renders the `BoilingVerdict` for the current input value.
+
+```js{5,10,15,19-22}
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.state = {celsius: 0};
+  }
+
+  handleChange(e) {
+    this.setState({
+      celsius: parseFloat(e.target.value, 10)
```

Comment:
Small nitpick, but I don't think [parseFloat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat) takes a radix, only [parseInt()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt), so I believe the *base10* is ignored. This radix is also passed is in a few other places throughout this code.